untitled/main/com/aca/homework/week14/thread/interrupt/InterruptionSensitiveRunnable.java

```diff
@@ -0,0 +1,32 @@
+package com.aca.homework.week14.thread.interrupt;
+
+import java.util.List;
+import java.util.UUID;
+
+import static java.lang.Thread.sleep;
+
+public class InterruptionSensitiveRunnable implements Runnable {
+
+    private final List<String> sharedList;
+
+    public InterruptionSensitiveRunnable(List<String> sharedList) {
+        this.sharedList = sharedList;
+    }
+
+    @Override
+    public void run() {
+        while (true) {
+            if (Thread.interrupted()) return;
+
+            synchronized (sharedList) {
```

Comment:
please pass the thread-safe list implementation instead of using the synchronized keyword. Because if you write using the synchronized you should read using the synchronized too.